services/venice-router/src/main/java/com/linkedin/venice/router/throttle/StoreReadThrottler.java

```diff
@@ -62,9 +62,9 @@ public StoreReadThrottler(
     }
   }
 
-  public void mayThrottleRead(double readCapacityUnit, Optional<String> storageNodeId) {
-    if (storageNodeId.isPresent()) {
-      EventThrottler storageNodeThrottler = storageNodesThrottlers.get(storageNodeId.get());
+  public void mayThrottleRead(double readCapacityUnit, String storageNodeId) {
+    if (storageNodeId != null) {
+      EventThrottler storageNodeThrottler = storageNodesThrottlers.get(storageNodeId);
```

Comment:
If `per storage node throttler` is disabled, we shouldn't need to build it at all, as Router doesn't need to construct `storageNodeThrottler` and no need to listen to partition assignment change.

Reply:
I agree, I left them knowingly to keep the code change smaller as once we decide on the config, we will delete the code.

Reply:
No, this is not about deleting the code right away.
We should pass this new config to `ReadRequestThrottler`, and when the per storage node throttler is disabled, we don't need to construct them and no need to listen the partition assignment change.
This is an efficiency improvement.